uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php

```diff
@@ -591,12 +591,96 @@ public function upgrade_1026() {
     return $this->upgrade_1020();
   }
 
-    public function uninstall()
-    {
-        CRM_Core_DAO::executeQuery("DELETE FROM `civicrm_navigation` WHERE name IN ('tasksassignments', 'ta_dashboard', 'tasksassignments_administer', 'ta_settings')");
-        CRM_Core_BAO_Navigation::resetNavigation();
+  /**
+   * Remove the old Dashboard submenu item
+   */
+  public function upgrade_1027() {
+    $taDashboard = new CRM_Core_DAO_Navigation();
+    $taDashboard->name = 'ta_dashboard';
+    $taDashboard->find(true);
```

Comment:
Just to make the upgraders idempotent do you think it makes sense to put the same

```
if (!$taDashboard->id) {
  return TRUE;
}
```

check as below in this one?